src/Compilers/CSharp/Test/Semantic/Semantics/AccessCheckTests.cs

```diff
@@ -1153,5 +1153,57 @@ public class TestClass1 { }
 ///////////////";
             CreateCompilationWithMscorlib(source).GetDiagnostics();
         }
+
+        [Fact, WorkItem(13652, "https://github.com/dotnet/roslyn/issues/13652")]
+        public void UnusedFieldInAbstractClassShouldTriggerWarning()
+        {
+            var text = @"
+abstract class Class1
+{
+    private int _UnusedField;
+}";
+            CompileAndVerify(text).VerifyDiagnostics(
+                // (4,21): warning CS0169: The field 'Class1._UnusedField' is never used
+                //         private int _UnusedField;
+                Diagnostic(ErrorCode.WRN_UnreferencedField, "_UnusedField").WithArguments("Class1._UnusedField").WithLocation(4, 17));
+        }
+
+        [Fact, WorkItem(13652, "https://github.com/dotnet/roslyn/issues/13652")]
+        public void AssignedButNotReadFieldInAbstractClassShouldTriggerWarning()
+        {
+            var text = @"
+abstract class Class1
+{
+    private int _AssignedButNotReadField;
+
+    public Class1()
+    {
+        _AssignedButNotReadField = 1;
+    }
+}";
+            CompileAndVerify(text).VerifyDiagnostics(
+                // (4,21): warning CS0414: The field 'Class1._AssignedButNotReadField' is assigned but its value is never used
+                //         private int _AssignedButNotReadField;
+                Diagnostic(ErrorCode.WRN_UnreferencedFieldAssg, "_AssignedButNotReadField").WithArguments("Class1._AssignedButNotReadField").WithLocation(4, 17));
+        }
+
+        [Fact, WorkItem(13652, "https://github.com/dotnet/roslyn/issues/13652")]
+        public void UsedButNotAssignedFieldInAbstractClassShouldTriggerWarning()
```

Comment:
>UsedButNotAssignedFieldInAbstractClassShouldTriggerWarning [](start = 20, length = 58)

It feels like it would be good to have a test for a scenario when the class is public, or there is an InternalsVisisbleTo attribute.